craft_providers/actions/snap_installer.py

```diff
@@ -81,9 +82,51 @@ def _pack_host_snap(*, snap_name: str, output: pathlib.Path) -> None:
     )
 
 
+def _get_host_snap_revision(snap_name: str) -> str:
+    """Get the revision of the snap on the host."""
+    quoted_name = urllib.parse.quote(snap_name, safe="")
+    url = f"http+unix://%2Frun%2Fsnapd.socket/v2/snaps/{quoted_name}"
+    try:
+        snap_info = requests_unixsocket.get(url)
+    except requests.exceptions.ConnectionError as error:
+        raise SnapInstallationError(
+            brief="Unable to connect to snapd service."
+        ) from error
+    snap_info.raise_for_status()
+    return snap_info.json()["result"]["revision"]
+
+
+def _get_target_snap_revision(snap_name: str, executor: Executor) -> Optional[str]:
+    """Get the revision of the snap on the target."""
+    config = InstanceConfiguration.load(executor=executor)
+    if config is not None:
+        data = config.marshal()
```

Comment:
Do you need this `marshal` call? you can access the info doing `data.snaps[snap_name]["revision"]`